text/0145-remove-unnecessary-allocator-usage.md

```diff
@@ -0,0 +1,1013 @@
+
+#### ext_default_child_storage_read
+
+##### Existing prototype
+
+```wat
+(func $ext_default_child_storage_read_version_1
+    (param $child_storage_key i64) (param $key i64) (param $value_out i64) (param $offset i32)
+    (result i64))
+```
+
+##### Changes
+
+The function was returning a SCALE-encoded `Option`-wrapped 32-bit integer representing the number of bytes left at supplied `offset`. It was using a host-allocated buffer to return it. It is changed to always return the full length of the value directly as a primitive value.
+
+##### New prototype
+
+```wat
+(func $ext_default_child_storage_read_version_2
+    (param $storage_key i64) (param $key i64) (param $value_out i64) (param $value_offset i32)
+    (result i64))
+```
+
+##### Arguments
+
+* `storage_key` is a pointer-size ([Definition 216](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size)) to the child storage key ([Definition 219](https://spec.polkadot.network/chap-host-api#defn-child-storage-type));
+* `key` is the storage key being read;
+* `value_out` is a pointer-size ([Definition 216](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size)) to a buffer where the value read should be stored. The value is actually stored only if the buffer is large enough. Otherwise, the buffer contents are undefined;
+* `value_offset` is a 32-bit offset from which the value reading should start.
+
+##### Result
+
+The result is an optional positive integer ([New Definition I](#new-def-i)), representing either the full length of the value in storage or the _absence_ of such a value in storage.
+
+#### ext_default_child_storage_storage_kill
+
+##### Existing prototype
+
+```wat
+(func $ext_default_child_storage_storage_kill_version_3
+    (param $child_storage_key i64) (param $limit i64)
+    (result i64))
+```
+
+##### Changes
+
+The function used to accept only a child storage key and a limit and return a SCALE-encoded `enum` representing the number of iterations performed, wrapped into a discriminator to differentiate if all the keys were removed. It was using a host-allocated buffer to return the value. As [discussed](https://github.com/w3f/polkadot-spec/issues/588), such implementation was suboptimal, and a better implementation was proposed in [PPP#7](https://github.com/w3f/PPPs/pull/7), but the PPP has never been adopted. The new version adopts the PPP, providing a means of returning much more exhaustive information about the work performed, and also accepts an optional input cursor and makes the limit optional as well. It always returns the full length of the continuation cursor.
+
+##### New prototype
+
+```wat
+(func $ext_default_child_storage_storage_kill_version_4
+    (param $storage_key i64) (param $maybe_limit i64) (param $maybe_cursor_in i64)
+    (param $maybe_cursor_out i64) (param $backend i32) (param $unique i32) (param $loops i32)
+    (result i32))
```

Comment:
Same conversation as we had with `ext_storage_clear_prefix_version_3`, and also similarly too many parameters.

This comment also applies to all other similar host calls below.